File: hashtable/1/1.js
var twoSum = function(nums, target) {
  let map = new Map();
  let result = [];
  for (let [i, num] of nums.entries()) {
    if (map.get(target - num) !== undefined) {
      return [i, map.get(target - num)];
    }

    map.set(num, i);
  }
};

File: hashtable/1/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsToIndex = {}
        for (i, num) in enumerate(nums):
            if target-num in numsToIndex:
                return [i, numsToIndex[target-num]]
            else:
                numsToIndex[num] = i
